File: R/8kyu/8kyu-calculate-average.r
# Calculate average
# Language: R
# Difficulty: 8 kyu

find_average <- function(vec){
  ifelse(length(vec) == 0, 0, mean(vec))
}


# Tests
test_that('Basic tests', {
  expect_equal(find_average(c(1,1,1)),1)
  expect_equal(find_average(c(1,2,3)),2)  
  expect_equal(find_average(c(1,2,3,4)),2.5)
  expect_equal(find_average(c()),0)  
})

File: Python/8kyu/8kyu-calculate-average.py
# Calculate average
# Language: Python
# Difficulty: 8 kyu

# Solution 1
def find_average(numbers):
    return 0 if len(numbers) == 0 else sum(numbers) / len(numbers)

# Refactored Solution 1
def find_average(numbers):
   return sum(numbers) / len(numbers)


# Tests
import codewars_test as test
from solution import find_average

@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(find_average([1, 2, 3]), 2)